### PR TITLE
Cabanaghost

### PR DESCRIPTION
**Pull request by patrickb314**
Added cabanaghost package to develop.

### CODE REVIEW

**Comment by patrickb314** [comment]
Accidentally created PR to spack, not the fork that this lives on.

File: var/spack/repos/builtin/packages/cabana/package.py [hunk 1]
```diff
@@ -104,6 +105,9 @@ class Cabana(CMakePackage, CudaPackage, ROCmPackage):
     # Cajita support requires MPI
     conflicts("+cajita ~mpi")
     conflicts("+grid ~mpi")
+    
+    # MPI Stream communication requires MPI
+    conflicts("+mpistream ~mpi")
```
Comment (junghans) [inline var/spack/repos/builtin/packages/cabana/package.py:110]: If this is a new option, we should conflict on old versions, e.g.:
```
conflicts("+mpistream @:0.6")
```

File: var/spack/repos/builtin/packages/cabanaghost/package.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    # CMake specific build functions
+    def cmake_args(self):
+        args = []
+
+        # Use hipcc as the c compiler if we are compiling for rocm. Doing it this way
+        # keeps the wrapper insted of changing CMAKE_CXX_COMPILER keeps the spack wrapper
+        # and the rpaths it sets for us from the underlying spec.
+        if "+rocm" in self.spec:
+            env["SPACK_CXX"] = self.spec["hip"].hipcc
```
Comment (junghans) [inline var/spack/repos/builtin/packages/cabanaghost/package.py:79]: Seems a bit hackish, is that really the way to go?

File: var/spack/repos/builtin/packages/cabanaghost/package.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    maintainers("patrickb314", "EvanDrakeSuggs")
+
+    license("BSD-3-Clause")
+
+    version("develop", branch="develop")
+    version("main", branch="main")
+
+    # Variants are primarily backends to build on GPU systems and pass the right
+    # informtion to the packages we depend on
+    variant("cuda", default=False, description="Use CUDA support from subpackages")
```
Comment (streeve) [inline var/spack/repos/builtin/packages/cabanaghost/package.py:24]: rocm variant?

File: var/spack/repos/builtin/packages/cabana/package.py [hunk 4]
```diff
@@ -114,7 +118,7 @@ class Cabana(CMakePackage, CudaPackage, ROCmPackage):
     def cmake_args(self):
         options = [self.define_from_variant("BUILD_SHARED_LIBS", "shared")]
 
-        enable = ["CAJITA", "TESTING", "EXAMPLES", "PERFORMANCE_TESTING"]
+        enable = ["CAJITA", "TESTING", "EXAMPLES", "PERFORMANCE_TESTING", "MPISTREAM"]
```
Comment (streeve) [inline var/spack/repos/builtin/packages/cabana/package.py:121]: Do we need changes in Cabana to support this? This won't currently be used

File: var/spack/repos/builtin/packages/cabanaghost/package.py (hunk 2 above)
Comment (streeve) [reply var/spack/repos/builtin/packages/cabanaghost/package.py:79]: We use `options.append(self.define("CMAKE_CXX_COMPILER", self.spec["hip"].hipcc))` (taken from the kokkos package)

**Comment by patrickb314** [comment]
Apologies for this PR. It was meant to be to an internal development fork of spack we are using for adding stream triggering support to Cabana and a simple halo exchange benchmark. It is not ready for integration or mainstreaming yet. I've closed the PR.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request closed without merging**